### ISSUE TITLE
Bug: mysql socket connection not possible with mysql_* modules

### ISSUE THREAD

**Issue by bjne** [ansible/ansible#15382]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION
- 2.0.1.0
- master
- Any?

There is a bug in module_utils/mysql.py that prevents socket connection from beeing possible due
to host beeing allways set, and is the default. The following patch (must be modified slightly for master)
fixes the issue:

```
--- ansible/module_utils/mysql.py.old   2016-04-12 13:11:40.406586851 +0200
+++ ansible/module_utils/mysql.py.new   2016-04-12 13:12:05.490691086 +0200
@@ -31,14 +31,13 @@

 def mysql_connect(module, login_user=None, login_password=None, config_file='', ssl_cert=None, ssl_key=None, ssl_ca=None, db=None, cursor_class=None):
     config = {
-       'host': module.params['login_host'],
-        'ssl': {
-            }
+        'ssl': {}
     }

     if module.params['login_unix_socket']:
         config['unix_socket'] = module.params['login_unix_socket']
     else:
+       config['host'] = module.params['login_host']
        config['port'] = module.params['login_port']

     if os.path.exists(config_file):

```

**Comment by bjne** [comment]
Pull request: https://github.com/ansible/ansible/pull/15385

**Comment by bjne** [comment]
Fix merged in ee18af48ef47281363556d282e0d7839c14264b5
